ArduSub/control_poshold.cpp

```diff
@@ -10,7 +10,7 @@
 bool Sub::poshold_init()
 {
     // fail to initialise PosHold mode if no GPS lock
-    if (!position_ok()) {
+    if (!position_ok() && (!visual_odom.enabled() || !visual_odom.healthy())) {
```

Comment:
I'm surprised that this direct check of visual odometry is required.  If the EKF is happy and knows it's position then I think the existing position_ok() check should be sufficient.

Reply:
I'm trying without it and I just can't get it into position hold. Did I miss something?
position_ok() for [sub](https://github.com/ArduPilot/ardupilot/blob/master/ArduSub/system.cpp#L224) seems to be the same as for [copter](https://github.com/ArduPilot/ardupilot/blob/master/ArduCopter/system.cpp#L296)

Reply:
Maybe the EKF origin hasn't been set?  If using MP then try right-mouse-button-clicking on the Data screen's map and Set Home >> Set EKF Origin and see if it shows up.  After that I suspect it will work.